python/cog/server/redis_queue.py

```diff
@@ -71,6 +72,7 @@ def __init__(
         self.predict_timeout = predict_timeout
         self.redis_db = redis_db
         self.report_setup_run_url = report_setup_run_url
+        self.max_failure_count = max_failure_count
```

Comment:
Did we want to check that it's `> 0` too?

Reply:
With the implementation here, 0 is a valid number. Negatives aren't, though they'll behave the same as 0.

`max_failure_count` is the maximum number of concurrent failures we'll accept before it exiss. So, 0 would mean exit after any failure. Perhaps it would be clearer renamed to `exit_after_failure_count`, and then making 1 mean exit after any failure (and making 0 invalid).

Reply:
Ah sorry, yes! Makes sense.